src/dash-table/DataTable.js

```diff
@@ -178,6 +178,45 @@ export const propTypes = {
             PropTypes.arrayOf(PropTypes.string)
         ]).isRequired,
 
+        number: PropTypes.shape({
+            presentation: PropTypes.oneOf(['input', 'dropdown']),
+            on_change: PropTypes.shape({
+                action: PropTypes.oneOf([
+                    'passthrough',
+                    'coerce',
+                    'validate'
+                ]),
+                failure: PropTypes.oneOf([
+                    'default',
+                    'prevent',
+                    'skip'
+                ])
+            }),
+            validation: PropTypes.shape({
+                allow_nan: PropTypes.bool,
+                allow_nully: PropTypes.bool
+            })
+        }),
+
+        text: PropTypes.shape({
+            presentation: PropTypes.oneOf(['input', 'dropdown']),
+            on_change: PropTypes.shape({
+                action: PropTypes.oneOf([
+                    'passthrough',
+                    'coerce',
+                    'validate'
+                ]),
+                failure: PropTypes.oneOf([
+                    'default',
+                    'prevent',
+                    'skip'
+                ])
```

Comment:
action/failure processing is identical for text -- hoping that the division in two steps + the validation options should be enough to cover all the flavoring without having to make custom lists by type